extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -199,6 +208,10 @@ export abstract class AzureAuth implements vscode.Disposable {
 		}
 		// Let's try to convert the access token type, worst case we'll have to prompt the user to do an interactive authentication.
 		const result = await this.refreshToken(tenant, resource, baseTokens.refreshToken);
+		// Verify that the user logged into this account
+		if (!this.verifyCorrectToken(result, account)) {
+			return undefined;
```

Comment:
I feel like we should be throwing errors in these cases so that callers can do something more useful about specific failures

Reply:
Kindly note that underlying getUserKey already throws when the key is not found who is already getting bubbled up from this function so throwing here would make it more consistent :)